app/src/main/java/org/glucosio/android/analytics/GoogleAnalytics.java

```diff
@@ -20,23 +23,36 @@ public void init(@NonNull final Context context) {
         mTracker = analytics.newTracker(BuildConfig.GOOGLE_ANALYTICS_TRACKER);
         mTracker.enableAdvertisingIdCollection(true);
 
+        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
+        enabled = sharedPref.getBoolean("pref_analytics_opt_in", false);
+
         if (BuildConfig.DEBUG) {
             com.google.android.gms.analytics.GoogleAnalytics.getInstance(context).setAppOptOut(true);
             Log.i("Glucosio", "DEBUG BUILD: ANALYTICS IS DISABLED");
+        } else {
+            if (!enabled) {
+                com.google.android.gms.analytics.GoogleAnalytics.getInstance(context).setAppOptOut(true);
```

Comment:
What is value by default? Should we always set it, just in case if the user enables it back